sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/impl/DependencyBuilders.scala

```diff
@@ -76,42 +44,56 @@ trait DependencyBuilders {
   }
 }
 
-final class ScalaJSGroupID private[sbtplugin] (groupID: String) {
-  def %%%(artifactID: String): ScalaJSGroupArtifactID = {
-    nonEmpty(artifactID, "Artifact ID")
-    new ScalaJSGroupArtifactID(groupID, artifactID, ScalaJSCrossVersion.binary)
-  }
-}
+final class ScalaJSGroupID private[sbtplugin] (private val groupID: String) {
+  def %%%(artifactID: String): CrossGroupArtifactID =
+    macro ScalaJSGroupID.auto_impl
 
-final class ScalaJSGroupArtifactID private[sbtplugin] (groupID: String,
-    artifactID: String, crossVersion: CrossVersion) {
-  def %(revision: String): ModuleID = {
-    nonEmpty(revision, "Revision")
-    ModuleID(groupID, artifactID, revision).cross(crossVersion)
-  }
+  def %%%!(artifactID: String): CrossGroupArtifactID =
+    ScalaJSGroupID.withCross(this, artifactID, ScalaJSCrossVersion.binary)
 }
 
-final class AutoGroupID private[sbtplugin] (groupID: String) {
-  def %?%(artifactID: String): AutoGroupArtifactID = {
+object ScalaJSGroupID {
+  import scala.reflect.macros.Context
+
+  /** Internal. Used by the macro implementing [[ScalaJSGroupID.%%%]]. Use:
+   *  {{{
+   *  ("a" % artifactID % revision).cross(cross)
+   *  }}}
+   *  instead.
+   */
+  def withCross(groupID: ScalaJSGroupID, artifactID: String,
+      cross: CrossVersion): CrossGroupArtifactID = {
     nonEmpty(artifactID, "Artifact ID")
-    new AutoGroupArtifactID(groupID, artifactID)
+    new CrossGroupArtifactID(groupID.groupID, artifactID, cross)
   }
-}
 
-final class AutoGroupArtifactID private[sbtplugin] (groupID: String,
-    artifactID: String) {
-  def %(revision: String): Def.Initialize[ModuleID] = {
-    nonEmpty(revision, "Revision")
+  def auto_impl(c: Context { type PrefixType = ScalaJSGroupID })(
+      artifactID: c.Expr[String]): c.Expr[CrossGroupArtifactID] = {
+    import c.universe._
+
+    // Hack to work around bug in sbt macros (wrong way of collecting local
+    // definitions)
```

Comment:
Steps to reproduce (last time we tried):
* Rename [`AutoImport`](https://github.com/scala-js/scala-js/blob/b4e00f5e0074a2afdba8f166f12af1aa2e0865ac/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala#L43) to `autoImport`
* Rename the alias [`autoImport`](https://github.com/scala-js/scala-js/blob/b4e00f5e0074a2afdba8f166f12af1aa2e0865ac/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala#L41) to `AutoImport`
* Change that macro to refer to `autoImport` rather than `AutoImport`